updog: upgrade signal-hook

Signal hook is at `signal-hook = "0.1.13"`, but the current version is `0.3`. When I tried the upgrade some breaking change occurred which I think is out-of-scope for the more mechanical version bumps that I am currently applying.

Specifically:

```
error[E0599]: no method named `cloned` found for struct `signal_hook::iterator::SignalsInfo` in the current scope
   --> updater/updog/src/main.rs:449:30
    |
449 |     let signals_bg = signals.cloned();
    |                              ^^^^^^ method not found in `signal_hook::iterator::SignalsInfo`
```

This should probably be tackled in isolation to ensure nothing about the API breaks.

--- a/sources/updater/updog/Cargo.toml
+++ b/sources/updater/updog/Cargo.toml
@@ -27,8 +27,7 @@ tough = { version = "0.10", features = ["http"] }
 update_metadata = { path = "../update_metadata" }
 structopt = "0.3"
 url = "2.1.0"
-# TODO - update signal-hook to the latest version. https://github.com/bottlerocket-os/bottlerocket/issues/1268
-signal-hook = "0.1.13"
+signal-hook = "0.3"
 models = { path = "../../models" }
 
 [dev-dependencies]

--- a/sources/updater/updog/src/main.rs
+++ b/sources/updater/updog/src/main.rs
@@ -12,7 +12,8 @@ use log::debug;
 use model::modeled_types::FriendlyVersion;
 use semver::Version;
 use serde::{Deserialize, Serialize};
-use signal_hook::{iterator::Signals, SIGTERM};
+use signal_hook::consts::SIGTERM;
+use signal_hook::iterator::Signals;
 use signpost::State;
 use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
 use snafu::{ErrorCompat, OptionExt, ResultExt};
@@ -435,10 +436,10 @@ fn output<T: Serialize>(json: bool, object: T, string: &str) -> Result<()> {
 
 fn initiate_reboot() -> Result<()> {
     // Set up signal handler for termination signals
-    let signals = Signals::new(&[SIGTERM]).context(error::Signal)?;
-    let signals_bg = signals.clone();
+    let mut signals = Signals::new(&[SIGTERM]).context(error::Signal)?;
+    let signals_handle = signals.handle();
     thread::spawn(move || {
-        for _sig in signals_bg.forever() {
+        for _sig in signals.forever() {
             // Ignore termination signals in case updog gets terminated
             // before getting to exit normally by itself after invoking
             // `shutdown -r` to complete the update.
@@ -450,7 +451,7 @@ fn initiate_reboot() -> Result<()> {
         .context(error::RebootFailure)
     {
         // Kill the signal handling thread
-        signals.close();
+        signals_handle.close();
         return Err(err);
     }
     Ok(())
@@ -506,11 +507,7 @@ fn main_inner() -> Result<()> {
     let arguments = parse_args(std::env::args());
 
     // SimpleLogger will send errors to stderr and anything less to stdout.
-    SimpleLogger::init(
-        arguments.log_level,
-        LogConfig::default(),
-    )
-    .context(error::Logger)?;
+    SimpleLogger::init(arguments.log_level, LogConfig::default()).context(error::Logger)?;
 
     let command =
         serde_plain::from_str::<Command>(&arguments.subcommand).unwrap_or_else(|_| usage());

